packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -0,0 +1,244 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { ReactNode, forwardRef, memo, useCallback, useState } from "react";
+import {
+  Box,
+  Chip,
+  ListSubheader,
+  MenuItem,
+  Select as MuiSelect,
+  SelectChangeEvent,
+} from "@mui/material";
+import { SelectProps as MuiSelectProps } from "@mui/material";
+import { Checkbox } from "./Checkbox";
+import { Field } from "./Field";
+
+export interface Option {
+  text: string;
+  value?: string;
+  type?: "heading" | "option" | string;
```

Comment:
I'm getting type errors on `optionsGrouped` in the `stories.tsx` file if I don't include `string` as a possible type. I'm not clear on the reason — it seems like it should be valid if it's `type?: "heading" | "option"` if those are the only two values I use, no?

Reply:
I put a comment for `optionsGrouped` to fix the type issue.